libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1911,8 +1911,26 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			break;
 		}
 		case Type::Category::Struct:
-			solUnimplementedAssert(false, "Struct conversion not implemented.");
+		{
+			solAssert(toCategory == Type::Category::Struct, "");
+			auto const& fromStructType = dynamic_cast<StructType const &>(_from);
+			auto const& toStructType = dynamic_cast<StructType const &>(_to);
+			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
+
+			solUnimplementedAssert(!fromStructType.isDynamicallyEncoded(), "");
+			solUnimplementedAssert(toStructType.location() == DataLocation::Memory, "");
+			solUnimplementedAssert(fromStructType.location() == DataLocation::CallData, "");
+
+			Whiskers t(R"(
+				converted := <abiDecode>(value, add(value, calldatasize()))
```

Comment:
```suggestion
				converted := <abiDecode>(value, calldatasize())
```

Either this is the end offset, then it's ``calldatasize()`` or it's the remaining length, then it's ``sub(calldatasize(), value)``, isn't it?

Reply:
It should be end offset, so will apply your suggested change

Reply:
It would be nice if violating this (i.e. trying to copy something that is too short in calldata) could be tested...
But I think it's rather hard, or probably even impossible, to come up with a suitable test case actually...
The second argument is used for a bounds check, but I think at this point here, we can already be sure that the calldata struct is in bounds, because it either stems from a function argument, which will have done bounds check when decoding its arguments, or from an index/member access of a larger calldata array/struct, but then that index/member access probably does the bounds check...
In general I could imagine that this will change a bit again, once the more complex part of dynamic calldata types is implemented (for which simply calling ``abiDecode`` won't be enough anymore)... But that's a different story...